### DISCUSSION TITLE
Sources urls should respect `ipv6_servers = false` option.

### DISCUSSION THREAD

**Discussion by vbauerster** [DNSCrypt/dnscrypt-proxy#2078] [Potential issues]
Would it be possible for the `sources` section to respect `ipv6_servers = false` option? I mean not to use ipv6 urls if `ipv6_servers = false`.

https://github.com/DNSCrypt/dnscrypt-proxy/blob/c467e20311ed21134d878a8f5ec230790e0b4355/dnscrypt-proxy/example-dnscrypt-proxy.toml#L670-L688

https://github.com/DNSCrypt/dnscrypt-proxy/blob/c467e20311ed21134d878a8f5ec230790e0b4355/dnscrypt-proxy/example-dnscrypt-proxy.toml#L64

**Comment by vbauerster** [comment]
I actually had a service crash most likely because of this (dnscrypt-proxy is self compiled from HEAD):
![Screen Shot 2022-03-28 at 20 34 30](https://user-images.githubusercontent.com/2206533/160575029-057be082-70bf-4869-825d-ddb9f98e3cd3.png)

**Comment by jedisct1** [comment]
If a source URL is unreachable, no matter why, the proxy is expected to try another URL in the list.

What exact error do you get when starting the proxy from the command line?

**Comment by vbauerster** [comment]
The problem was that service sometimes crashed. Now I figured out why. It's running on raspberry pi, where there is no hw clock. So if it is started before actual time is set, it always crashed with error saying something like `cert expired or invalid` at source url download step. I've fixed that by making sure dnscrypt-proxy is starting after actual time is set.

So if bad url is skipped, this issue is likely not so important anyway, feel free to close.

**Comment by jedisct1** [comment]
Unless the cache is stored in a location such as /tmp that doesn't persist after a reboot, the old cache should be used if up-to-date information cannot be fetched.

If nothing is accessible, it doesn't crash. The service stops, because it doesn't have any valid servers list.

In servers list, `https://` can be replaced with `http://` so that there's no TLS, no need for a correctly set clock. But the best thing to do is exactly what you did; make sure the time is set. Certificates, whether they are DNSCrypt or DoH, have a limited life time, so even if the configuration is usable, DNS servers won't be usable if the clock is off.

Waiting for the system clock not be in 1970's is something we could add to the netprobe step. though.
